core/src/main/java/com/linecorp/armeria/server/decoding/DecodingService.java

```diff
@@ -58,7 +60,10 @@
      */
     public static Function<? super HttpService, DecodingService> newDecorator(
             Iterable<? extends StreamDecoderFactory> decoderFactories) {
-        return delegate -> new DecodingService(delegate, decoderFactories);
+        requireNonNull(decoderFactories, "decoderFactories");
+        final Iterable<? extends StreamDecoderFactory> immutableDecoderFactories =
```

Comment:
- How about just using `List<...>`?
- The constructor could require an `ImmutableList` to statically ensure the constructor will always get an immutable list (i.e. a dev will be sure he/she doesn't need to worry about immutability of the given list).

Reply:
It'd be nice if you could make the same changes to `DecodingClient`. :bow:

Reply:
- change Iterable to List
- add `requireNonNull` & ImmutableList to `DecodingClient` (https://github.com/line/armeria/pull/3087/commits/09342ef1987f0bef9e0fd25376437fd69b11e9bc)